### PR TITLE
Added a check in the makefile for the latest conda build.

### LINKED ISSUE #21224: Useful errors when conda is out-of-date
## Reason
When there is a submodule update (and subsequent conda package update in MOOSE), there is no hint that conda is out-of-date when the user attempts to build after updating MOOSE only. This can lead to pain and confusion for users. 

We already perform a submodule mismatch check, and print out a warning during build. We should do the same for conda. 

## Design
We should take inspiration from `moose.mk`:
https://github.com/idaholab/moose/blob/0ffe4cbff032b2fdcb9a748a210ba6fd6327c141/framework/moose.mk#L332-L340

`conda_libmesh_status`, for example, should get the result of a query to conda to get the `moose-libmesh` version currently installed in the environment. Then the returned version should be compared to the version in `conda/libmesh/meta.yaml`. If there is a mismatch, a warning message should be reported to the user.

## Impact
Less confusion for users when MOOSE is updated and fails to build suddenly when conda isn't also updated.

### PR DESCRIPTION
**Pull request by socratesgorilla**
Closes #21224

### CODE REVIEW

**Review by cticenhour — CHANGES_REQUESTED** [review]
Some logic should be added to your changes that moves on if conda isn't found. We don't require that anyone uses our conda packages, so `conda list` should only be called if we know conda is available. 

I am also looking into what, if any, we can use from `scripts/verify_conda_libmesh.py`. I am under the impression that it performed this kind of check for those who had the conda build environment installed (to build local copies of packages), but it seems it has been broken for some time.

File: framework/moose.mk
Comment (socratesgorilla) [reply framework/moose.mk:15]: I put them up here because I assumed we wanted them before the check for Conda verification which directly follows them.

File: framework/moose.mk
Comment (socratesgorilla) [reply framework/moose.mk:16]: Wasn't aware of awk, and I was concerned about using -f on cut.

**Comment by socratesgorilla** [comment]
I didn't even notice the py script until you pointed it out, I'm not sure what's going on here either

File: framework/moose.mk
Comment (cticenhour) [reply framework/moose.mk:15]: That conda verification is currently skipped by @milljm, as it was intended for conda package maintainers early on in our package creation cycle. It's been left behind so that we might use it again later.

Regardless, I see your logic - don't mind you leaving it here.

**Comment by cticenhour** [comment]
I added some discussion about it - It's a legacy script we used two years ago. Might be some sense in reviving it, but it leans heavily on the conda-build package, which we don't necessarily want users to have to install to get these notices.

File: framework/moose.mk
Comment (cticenhour) [inline framework/moose.mk:17]:
```suggestion
check_conda := $(shell conda list &> /dev/null; echo $$?)
ifeq ($(check_conda),0)
```

The `.` isn't necessary here, and we want both stdout and stderr to be suppressed when this is called. I'd also like to use the explicit exit code 0 for the check rather than a string. Another exit code could indicate something is wrong with their conda installation, even if `conda list` executes.

**Review by cticenhour — CHANGES_REQUESTED** [review]
Only a couple tweaks to the conda check itself. This looks good otherwise.

File: framework/moose.mk
Comment (loganharbour) [reply framework/moose.mk:17]: No. We want stderr to be suppressed (because we don't care to parse the error) and stderr to be captured (which we need to parse).

Comment (cticenhour) [reply framework/moose.mk:17]: We don't need to parse stdout here. We need the exit code for the process. Parsing stdout for this command comes later.

Comment (loganharbour) [reply framework/moose.mk:17]: Yeah and we don't want to run conda list three times each time make is ran. It's slow. I talked to Nathaniel about this in person.

Comment (loganharbour) [reply framework/moose.mk:17]: `conda list` on my machine takes 0.5s, so 1.5s or greater if slow IO. No bueno.

Comment (cticenhour) [reply framework/moose.mk:17]: After talking offline, I understand what @loganharbour was referring to here. We should be caching the output of conda list at the same time we call it, so that we don't have to run it three times. I still would like to use some sort of exit code and not a failure string to indicate more subtle issues though. 

@socratesgorilla how do you feel about making a bash script to call here instead so that we can avoid some of the complexities of make? Logan and I both think the number of new lines could be shortened going that route. Then we might remove the old python script altogether. (As @milljm previously managed that, I would welcome his input on that last bit)

Comment (milljm) [reply framework/moose.mk:17]: Feel free to remove the verify_conda_libmesh.py, as I think it serves no purpose anymore. It is to bad that interfacing with their API is slow!

**Comment by cticenhour** [comment]
@socratesgorilla haven't forgotten about this, just wanted to run some local testing with the new changes. @loganharbour - does this need to be rebased on devel to test the new build env changes too?

**Review by cticenhour — APPROVED** [review]

File: framework/moose.mk
Comment (cticenhour) [inline framework/moose.mk:17]:
```suggestion
check_conda := $(shell conda list 2> /dev/null | grep "moose-libmesh \|moose-petsc"; echo 0)
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
Missed this one - we do still want to suppress the error, since we're not concerned with parsing it if it happens, and it might provide confusing output.

**Review by cticenhour — DISMISSED** [review]

**Comment by cticenhour** [comment]
Will merge once issues with next are fixed up.

**Comment by socratesgorilla** [comment]
Did next get cleaned? Is this mergeable now? @cticenhour

**Comment by cticenhour** [comment]
@milljm if I remember our conversation on this correctly, this kind of information might not be useful in a status .yaml file for an installed package (which I think was the intent of #20729?)

**Comment by cticenhour** [comment]
@socratesgorilla how much more effort would it take to also perform a build number check on top of the version? This would allow us to catch the case when we're making serious changes to a petsc or libmesh configuration that doesn't result from an mpich change. Your changes are adequate for the latter case (which is why I approved it), but @milljm reminded me of the former case that doesn't happen nearly as often.

**Review by cticenhour — APPROVED** [review]

**Pull request merged**
